### ISSUE TITLE
Target Different Deployments

### ISSUE THREAD

**Issue by ryanjbaxter** [dekorateio/dekorate#287]
It will be fairly common for an app to be deployed to various k8s deployments.  Depending on the deployment it might be necessary to change the manifest properties.  It would be nice if we could support generating manifests for any number of deployments.

**Comment by iocanel** [comment]
I am not sure I 100% get what do you mean by `deployments` are you refering to the Deployment resource, or the deployment platform?

Can you please provide an example?

**Comment by ryanjbaxter** [comment]
For example for development locally I might target Minikube or Docker Desktop, but for staging/production I might target GKE

**Comment by iocanel** [comment]
So basically we are talking about a profile handling mechanism righ?

**Comment by ryanjbaxter** [comment]
Yeah I guess so.  The term profile is overloaded, but essentially I want to have configuration for different k8s deployments I might be deploying my app to.

**Comment by iocanel** [comment]
At the moement, we have cases where we generate different stuff for different kubernetes distros  (kubernetes, openshift) and we could possibly add more if it makes sense.

What we completely lack is distinguishing between environment kinds (e.g. "staging", "production" etc).

So I guess that the first step would be to introudce a new property to the core annotations say `classifier` which is going  to be use pretty much like a maven classifier  (e.g. to generate `kubernetes-stagging.yml` and so on.

Next, step would be to add an array wrapper annotations so that we can define multple `@KubernetesApplication` annotations on a single class (one per classifier).

And last would be to support that in the annotationless mode via yaml or properties.
